src/test/java/space/obminyashka/items_exchange/end2end/AdvertisementFlowTest.java

```diff
@@ -128,6 +128,25 @@ void findPaginatedByCategoryId_shouldReturnPageResponse() throws Exception {
                 .andExpect(jsonPath("$.numberOfElements").value(advertisementRepository.count()));
     }
 
+    @Test
+    @WithMockUser(username = "user")
+    @DataSet("database_init.yml")
+    void findPaginatedAdvertisementWithSameSubcategory_shouldReturnPageProperQuantityOfAdvertisementWithoutRequestAdvertisement() throws Exception {
+        UUID advertisementId = UUID.fromString("65e3ee49-5927-40be-aafd-0461ce45f295");
+        Long subcategoryId = 1l;
+        sendUriAndGetResultAction(get(ADV_SUBCATEGORY_RANDOM, advertisementId, subcategoryId), status().isOk())
+                .andExpect(jsonPath("$.size()").value(advertisementRepository.countByIdNotAndSubcategoryId(advertisementId, subcategoryId)));
+    }
+
+    @Test
+    @WithMockUser(username = "user")
+    void findPaginatedAdvertisementWithNotExistedSubcategory_shouldReturnEmptyPage() throws Exception {
+        UUID advertisementId = UUID.fromString("65e3ee49-5927-40be-aafd-0461ce45f295");
+        Long subcategoryId = 4l;
+        sendUriAndGetResultAction(get(ADV_SUBCATEGORY_RANDOM, advertisementId, subcategoryId), status().isOk())
+                .andExpect(jsonPath("$.size()").value(0));
+    }
+
```

Comment:
That test doesn't use a dataset so it can be changed to a unit test and moved space.obminyashka.items_exchange.controller.AuthorizationControllerTest